### PR TITLE
Fixes #8405 - Filter :interfaces_attributes when calculating templates_used

### PR DESCRIPTION
**Pull request by GregSutcliffe**
This _might_ also affect the Host.new in #process_taxonomy too, but I've not tested that.

### CODE REVIEW

**Comment by GregSutcliffe** [comment]
Per the description in the ticket, I've dropped the JS side, it was too hard to arrange. It'll be refactored in the networking branch for 1.8 anyway.

**Comment by GregSutcliffe** [comment]
A quick test shows that the same issue and same fix is present in #process_taxonomy, so I've added that to the commit.

**Comment by GregSutcliffe** [comment]
Whoops, updated tests to not rely on things that are only in the network branch :)

File: test/functional/hosts_controller_test.rb [hunk 1]
```diff
@@ -833,6 +833,25 @@ class Host::Valid < Host::Base; end
...
+    xhr :put, :template_used, {:provisioning => 'build', :host => attrs }, set_session_user
+    assert_response :success
+    assert_template :partial => '_provisioning'
+  end
+
+  test 'process_taxonomy renders a host from the params correctly' do
+    nic=FactoryGirl.create(:nic_managed, :host => @host)
+    attrs = @host.attributes
+    attrs[:interfaces_attributes] = nic.attributes
+    xhr :put, :template_used, {:provisioning => 'build', :host => attrs }, set_session_user
```
Comment (ares) [inline test/functional/hosts_controller_test.rb]: should probably test process_taxonomy action?

**Comment by ares** [comment]
Tested, fixes the issue. @GregSutcliffe I think that the second test should be updated, then good to merge.

File: test/functional/hosts_controller_test.rb (hunk 1 above)
Comment (GregSutcliffe) [inline test/functional/hosts_controller_test.rb]: Ha, oops

**Comment by GregSutcliffe** [comment]
@ares updated, pending Jenkins :)

**Comment by ares** [comment]
Merged as https://github.com/theforeman/foreman/commit/be4602e87f8a8f35f49c8568fdea41e1570e5bb2, thanks @GregSutcliffe

**Pull request closed without merging**